components/builder-core/src/build_config.rs

```diff
@@ -156,6 +158,9 @@ pub struct ProjectCfg {
     /// notification to determine if an automatic rebuild should occur.
     #[serde(default)]
     pub paths: Vec<Pattern>,
+    /// Package targets to build when changes detected
+    #[serde(default = "ProjectCfg::default_build_targets")]
+    pub build_targets: HashSet<PackageTarget>,
```

Comment:
Do we need some error handling around the reading of `build_targets`?  I would expect that typos and experimentation would be common with this setting.  Changing the tests below to an invalid PackageTarget  results in the following error: 

```
---- build_config::test::triggered_by stdout ----
thread 'build_config::test::triggered_by' panicked at 'called `Result::unwrap()` on an `Err` value: DecryptError("string cannot be parsed: \"i-am-not-a-package-target\" (Package targets must be in architecture-platform format (example: x86_64-linux)) for key `builder-api.build_targets`")', src/libcore/result.rs:1009:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.
```

Would that cause builder-api to crash if a user connected a repo with a malformed build_target? or would it fail to load the the .bldr.toml file for that project resulting in no builds for the user.  

Is there a way we could provide feedback to the user to let them know?

Reply:
That is a really great question. Currently, if we fail to correctly parse the bldr.toml, we just fall back to default values (ie, pretend that the bldr.toml file doesnt exist).  There is no panic, but we do drop the error on the floor (other than logging a debug output).  Instead of continuing on this way, we could fail the push request entirely with an error code and some text. Unfortunately that still isn't really easily visible to the user - you have to go to the specific github app settings to find the result of the push, and that's not terribly intuitive either.